cluster/cluster.go

```diff
@@ -49,6 +53,10 @@ type SegConfig struct {
 	DbID      int
 	ContentID int
 	Role      string
+	PRole     string
```

Comment:
I'm not sure if introducing these properties will break backwards compatibility or cause any other dependency issues ?

Reply:
I doubt it will do, the lib will be used at compile time to create a binary. but for safer side I will try to notify about the pr on related slack channels.

Reply:
What can happen is that existing utilities calling the other function `GetSegmentConfiguration` will now get the new struct but the newly added properties won't be set since they are only set in `GetSegmentConfigurationFromFile`.  It's possible that a utility might try to access one of these properties and error out or worse have unintended side effects.

Is there a reason we decided to not initialize these properties while calling `GetSegmentConfiguration` as well?

Reply:
don't know what the purpose of not having all the SegConfig members the same as the gp_segment_configuration table. one thing I can think of the missing fields are ok to have nil value while using gpdbArray. 

your observation is valid that the user may accidentally access the new member assuming that it may have valid data but can find nil or empty. 

IMO 2 ways we can handle this issue 
1. keeping the struct the same as the previous and ignoring the missing member in GetSegmentConfigurationFromFile API 
2.  Update the GetSegmentConfiguration() API  to populate newly added members. 

Please provide your opinion, both the approaches looks ok to me.

Reply:
I think option 2 is right way to go but it would be nice if someone else also takes a look. What do you think @Chibin  ?
Also I'm not sure what option 1 means and how it solves the problem ?

Reply:
I think that... at the very least, we should do option 2.